### ISSUE TITLE
Staging Review finding: Unexpected behavior for `All` checkboxes

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#126948]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** govcio-vebt
**Product:** Comparison Tool
**Feature:** Student feedback grid

## Findings details
**VA.gov Experience Standard - issue:** [User experiences an unexpected interaction with a component or pattern.](https://design.va.gov/about/experience-standards/#user-experiences-predictable-interactions-with-components)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @erinrwhite (IA), @allison0034 (Design), @jasonday (Accessibility)

### Description

In the filter accordions, there are two sets of checkboxes. For each set, there is an "All" option. When selecting the All option, the expected behavior would be that it automatically checks all of the checkboxes below (to indicate they are included in the search). This doesn't happen, which makes it appear that "All" is a separate and distinct option.

<details><summary>Link, screenshot or steps to recreate</summary>

to be added

</details>

## Recommended action

Use the [checkbox - indeterminate](https://design.va.gov/components/form/checkbox#indeterminate) component for this interaction, indicating that the `All` checkbox selects everything below as well.

## References

- **Accessibility Defect Severity:** N/A
- **WCAG Success Criteria:** N/A
- **Modality:** N/A
- **Design System Component:** [Form - Checkbox component](https://design.va.gov/components/form/checkbox)
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A
- **VADS Guidance:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://ows.io/qs/o3jkwoez) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue; or if you believe the finding should not be fixed; comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.
